tests/test_dynamic_groups.c

```diff
@@ -46,14 +58,23 @@ int main(void)
 
 	struct rtr_mgr_config *conf;
 
-	rtr_mgr_init(&conf, groups, 1, 30, 600, 600, NULL, NULL, NULL, NULL);
+	rtr_mgr_init(&conf, groups, 1, 30, 600, 600, NULL, NULL, &connection_status_callback, NULL);
 
 	//start the connection manager
 	rtr_mgr_start(conf);
 
-	//wait till at least one group is fully synchronized with the server
-	while (!rtr_mgr_conf_in_sync(conf))
+	int sleep_counter = 0;
+	// wait 20 sec till at least one group is fully synchronized with the server
+	// otherwise EXIT_FAILURE.
+	while (!rtr_mgr_conf_in_sync(conf)) {
+		if (connection_status == RTR_MGR_ERROR)
```

Comment:
maybe nit picky, but I would write this as:

```C
while (!rtr_mgr_conf_in_sync(conf)) {
    sleep_counter++;
    if ((connection_status == RTR_MGR_ERROR) ||
        (sleep_counter > connection_timeout))
            return EXIT_FAILURE;
    sleep(1);
}
```

as the sleep comes last this should only sleep 20 times at most, or exit before, thus avoiding the `>=` and also merging both error conditions.